packages/create-pantheon-decoupled-kit/src/templates/next-drupal-app/app/[lang]/layout.tsx

```diff
@@ -0,0 +1,73 @@
+import '@pantheon-systems/nextjs-kit/style.css';
+
+import { Footer } from '@pantheon-systems/nextjs-kit/dist/components/footer';
+import { Header } from '@pantheon-systems/nextjs-kit/dist/components/header';
+import styles from './layout.module.css';
+import {
+	getCurrentLocaleStore,
+	globalDrupalStateStores,
+} from '../../lib/stores';
+
+import './globals.css';
+
+type FooterData = {
+	title: string,
+	url: string,
+	parent: string
+}
+
+async function getData(locale: string) {
+	const store = getCurrentLocaleStore(locale, globalDrupalStateStores);
+	const footerMenu = await store.getObject({
+		objectName: 'menu_items--main',
+		anon: true,
+	});
+
+  return footerMenu as FooterData[];
```

Comment:
No need to typecast, `getObject` supports a generic which should spit the data out as that type (see suggestion above).